uui/components/datePickers/RangeDatePickerBody.tsx

```diff
@@ -148,116 +149,148 @@ export class RangeDatePickerBody extends React.Component<RangeDatePickerBodyProp
+            newValue = {
+                ...newValue,
+                view: value.view,
+            };
+        }
 
-        this.props.onValueChange({
-            selectedDate: this.props.value.selectedDate,
-            displayedDate: part === 'from' ? displayedDate : displayedDate.subtract(1, 'month'),
-            view: view,
+        setActiveMonth(part);
+        props.onValueChange({
+            ...props.value,
+            ...newValue,
         });
-    }
+    };
 
-    getFromValue = (): PickerBodyValue<string> => {
-        return {
-            ...this.props.value,
-            view: this.state.activePart === 'from' ? this.props.value.view : 'DAY_SELECTION',
-            selectedDate: this.props.value.selectedDate?.from || null,
-        };
+    const renderDay = (renderProps: DayProps): JSX.Element => {
+        return (
+            <Day
+                { ...renderProps }
+                cx={ getDayCX(renderProps.value, props.value.selectedDate) }
+            />
+        );
     };
 
-    getToValue = (): PickerBodyValue<string> => {
-        return {
-            view: this.state.activePart === 'to' ? this.props.value.view : 'DAY_SELECTION',
-            displayedDate: this.props.value.displayedDate.add(1, 'month'),
-            selectedDate: this.props.value.selectedDate?.from || null,
-        };
+    const from: PickerBodyValue<string> = {
+        ...props.value,
+        view: activeMonth === 'from' ? props.value.view : 'DAY_SELECTION',
+        selectedDate: null,
     };
 
-    renderPresets = (presets: RangeDatePickerPresets) => {
+    const to: PickerBodyValue<string> = {
+        view: activeMonth === 'to' ? props.value.view : 'DAY_SELECTION',
+        month: props.value.month.add(1, 'month'),
+        selectedDate: null,
+    };
+
+    const renderPresets = (presets: RangeDatePickerPresets) => {
         return (
-            <>
+            <React.Fragment>
                 <div className={ uuiRangeDatePickerBody.separator } />
                 <CalendarPresets
-                    forwardedRef={ this.props.forwardedRef }
                     onPresetSet={ (presetVal) => {
-                        this.props.onValueChange({
+                        props.onValueChange({
+                            ...props.value,
                             view: 'DAY_SELECTION',
-                            selectedDate: { from: dayjs(presetVal.from).format(valueFormat), to: dayjs(presetVal.to).format(valueFormat) },
-                            displayedDate: dayjs(presetVal.from),
+                            selectedDate: {
+                                from: dayjs(presetVal.from).format(valueFormat),
+                                to: dayjs(presetVal.to).format(valueFormat),
+                            },
+                            month: dayjs(presetVal.from),
                         });
-                        this.props.changeIsOpen?.(false);
+                        // toggleIsOpen(false);
```

Comment:
How we are gonna close the body? Currently, we closing it only when `to` changed and both dates filled. Should we leave close callback in props or move renderPresets to a props? @AlekseyManetov

Reply:
I guess we can leave current behavior and close picker according the our global logic in `props.onValueChange` callback.